test/integration/restart_task_test.rb

```diff
@@ -36,7 +36,8 @@ def test_restart_by_annotation
   end
 
   def test_restart_statefulset_on_delete_restarts_child_pods
-    result = deploy_fixtures("hello-cloud", subset: "stateful_set.yml") do |fixtures|
+    result = deploy_fixtures("hello-cloud", subset: ["configmap-data.yml", "unmanaged-pod-1.yml.erb",
```

Comment:
Deploying a bare pod in the integration test without the fix produces the expected error:

```
Error:
RestartTaskTest#test_restart_statefulset_on_delete_restarts_child_pods:
NoMethodError: undefined method `find' for nil:NilClass
    /home/runner/work/krane/krane/lib/krane/restart_task.rb:216:in `block in delete_statefulset_pods'
    /opt/hostedtoolcache/Ruby/3.0.3/x64/lib/ruby/3.0.0/delegate.rb:349:in `select!'
    /opt/hostedtoolcache/Ruby/3.0.3/x64/lib/ruby/3.0.0/delegate.rb:349:in `block in delegating_block'
    /home/runner/work/krane/krane/lib/krane/restart_task.rb:215:in `delete_statefulset_pods'
    /home/runner/work/krane/krane/lib/krane/restart_task.rb:239:in `block in restart_statefulsets!'
    /home/runner/work/krane/krane/lib/krane/restart_task.rb:233:in `each'
    /home/runner/work/krane/krane/lib/krane/restart_task.rb:233:in `restart_statefulsets!'
    /home/runner/work/krane/krane/lib/krane/restart_task.rb:74:in `run!'
    /home/runner/work/krane/krane/lib/krane/restart_task.rb:49:in `run'
    /home/runner/work/krane/krane/test/integration/restart_task_test.rb:49:in `test_restart_statefulset_on_delete_restarts_child_pods'
```